federation/apis/federation/types.go

```diff
@@ -153,3 +153,16 @@ type ClusterReplicaSetPreferences struct {
 	// A number expressing the preference to put an additional replica to this LocalReplicaSet. 0 by default.
 	Weight int64
 }
+
+// Annotations for a federated service to keep record of service load balancer ingress endpoints
```

Comment:
- storing the service ingress information in annotation of federated service rather than cache will prevent loss of information if at all the service controller restarts and also aids in decoupling features which can independently act on this information like the dns_controller. (also hoping, in future this information can directly be used to write a back-end for CoreDNS and eliminate the need for separate etcd instance for CoreDNS).
- cluster service status in the service object does not hold the information about whether backing endpoint is healthy.
- storing the information about cluster name to which ingress belongs helps in multiple places.